### PR TITLE
[Kernel Slimming] Migrate AWQ marlin repack kernel to JIT

### PR DESCRIPTION
**Pull request by celve**
## Motivation

See https://github.com/sgl-project/sglang/issues/17865

## Modifications

New files:
- python/sglang/jit_kernel/csrc/gemm/marlin/awq_marlin_repack.cuh — JIT-compiled CUDA kernel ported from sgl-kernel/csrc/gemm/marlin/awq_marlin_repack.cu
- python/sglang/jit_kernel/awq_marlin_repack.py — Python wrapper with JIT loading and output tensor allocation
- python/sglang/jit_kernel/tests/test_awq_marlin_repack.py
- python/sglang/jit_kernel/benchmark/bench_awq_marlin_repack.py

Modified files:
- python/sglang/srt/layers/quantization/awq.py — Switch awq_marlin_repack import from sgl_kernel to sglang.jit_kernel

## Accuracy Tests

Pass all tests defined in python/sglang/jit_kernel/tests/test_awq_marlin_repack.py

## Benchmarking and Profiling

```
awq-marlin-repack-performance:
   size_k  JIT Kernel  AOT Kernel
0   512.0   12.126000   12.121699
1  1024.0   12.187091   15.707735
2  2048.0   23.288894   30.660015
3  4096.0   51.975607   58.915189
4  8192.0  103.436129  103.417934
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/awq.py [hunk 1]
```diff
@@ -60,7 +60,9 @@
     import torch_npu
 
 if _is_cuda:
-    from sgl_kernel import awq_dequantize, awq_marlin_moe_repack, awq_marlin_repack
+    from sgl_kernel import awq_dequantize, awq_marlin_moe_repack
```
Comment (BBuf) [inline python/sglang/srt/layers/quantization/awq.py]: todo: we should also remove `awq_dequantize` and `awq_marlin_moe_repack` to jit_kernel

**Review by BBuf — APPROVED** [review]
